### PR TITLE
Updating code to reuse ingress-gce/pkg/backends.ServicePort

### PR DESCRIPTION
**Pull request by nikhiljindal**
This is required for URLMaps. ServicePort also contain information about the corresponding service.
Updated the code to use the new fields to generate a better description for GCP resources (only health check for now)

Removed the custom ServicePort struct.

cc @csbell @G-Harmon

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Verified that all tests pass

```
ok      github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/cmd     0.333s
?       github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/gcp/cloudinterface  [no test files]
ok      github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/gcp/healthcheck     0.210s
ok      github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/gcp/loadbalancer    0.336s
?       github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/gcp/namer   [no test files]
```

**Comment by nikhiljindal** [comment]
Will update https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/3 to use the updated ServicePort as well once this merges.

File: app/mci/pkg/gcp/healthcheck/healthchecksyncer.go [hunk 1]
```diff
@@ -21,10 +21,10 @@ import (
 	"github.com/golang/glog"
 	"github.com/hashicorp/go-multierror"
 	compute "google.golang.org/api/compute/v1"
+	ingressbs "k8s.io/ingress-gce/pkg/backends"
```
Comment (G-Harmon) [inline app/mci/pkg/gcp/healthcheck/healthchecksyncer.go]: why ingressbs? Is that auto-selected? Seems like "ingressb" or "ingressbe" would be more inline with "ingresshc" below.

File: app/mci/pkg/gcp/healthcheck/healthchecksyncer.go [hunk 2]
```diff
@@ -109,11 +109,11 @@ func healthCheckMatches(desiredHC, existingHC *compute.HealthCheck) bool {
 	return true
 }
 
-func (h *HealthCheckSyncer) desiredHealthCheck(lbName string, port sp.ServicePort) (compute.HealthCheck, error) {
+func (h *HealthCheckSyncer) desiredHealthCheck(lbName string, port ingressbs.ServicePort) (compute.HealthCheck, error) {
 	// Compute the desired health check.
 	hc := compute.HealthCheck{
 		Name:        h.namer.HealthCheckName(port.Port),
-		Description: fmt.Sprintf("Kubernetes L7 Loadbalancing health check for multi cluster ingress %s.", lbName),
+		Description: fmt.Sprintf("Health check for service %s as part of kubernetes multi cluster loadbalancer %s", port.Description(), lbName),
```
Comment (G-Harmon) [inline app/mci/pkg/gcp/healthcheck/healthchecksyncer.go]: jonathan was asking about "multicluster" spelling, and we said it was without a space.

File: app/mci/pkg/gcp/healthcheck/fake_healthchecksyncer.go [hunk 3]
```diff
@@ -15,12 +15,12 @@
 package healthcheck
 
 import (
-	sp "github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/mci/pkg/serviceport"
+	ingressbs "k8s.io/ingress-gce/pkg/backends"
```
Comment (csbell) [inline app/mci/pkg/gcp/healthcheck/fake_healthchecksyncer.go]: ingress b.s. ? ;-) how about ingressbes ? or ingbesvc ...

File: app/mci/pkg/gcp/healthcheck/healthchecksyncer.go (hunk 2 above)
Comment (nikhiljindal) [reply app/mci/pkg/gcp/healthcheck/healthchecksyncer.go]: Removed the space

File: app/mci/pkg/gcp/healthcheck/fake_healthchecksyncer.go (hunk 3 above)
Comment (nikhiljindal) [reply app/mci/pkg/gcp/healthcheck/fake_healthchecksyncer.go]: Updated to ingressbe

File: app/mci/pkg/gcp/healthcheck/healthchecksyncer.go (hunk 1 above)
Comment (nikhiljindal) [reply app/mci/pkg/gcp/healthcheck/healthchecksyncer.go]: Updated to ingressbe

**Comment by nikhiljindal** [comment]
Updated code as per comments.
PTAL

**Pull request merged**